### PR TITLE
Parse IPv6 fix header

### PR DESCRIPTION
**Pull request by mazdakn**
## Description
This PR parses IPv6 fix header (not extensions yet).

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf-gpl/parsing.h [hunk 1]
```diff
@@ -10,6 +10,66 @@
 #define PARSING_ALLOW_WITHOUT_ENFORCING_POLICY 2
 #define PARSING_ERROR -1
 
+#define MAX_EXTENSIONS 2
+
+static CALI_BPF_INLINE int parse_packet_ipv6(struct cali_tc_ctx *ctx, int iterations) {
+	__u8 next_header = ipv6_hdr(ctx)->nexthdr;
+
+	for (int i = 0; i < iterations; i++) {
```
Comment (tomastigera) [inline felix/bpf-gpl/parsing.h:18]: do you know the value upfront or is it just a static max decided at compile time? Does it need to be provided from outside?

File: felix/bpf-gpl/parsing.h [hunk 2]
```diff
@@ -10,6 +10,66 @@
...
+		next_header = ipv6ext_hdr(ctx)->nexthdr;
+	}
+	CALI_DEBUG("Too many IPv6 extension headers\n");
+	// TODO: Jump to the parser program to extract more headers
+
+deny:
+	CALI_DEBUG("Failed to parse IPv6 header\n");
+	return PARSING_ERROR;
+
+allow_no_fib:
```
Comment (tomastigera) [inline felix/bpf-gpl/parsing.h:65]: no fib?

File: felix/bpf-gpl/parsing.h [hunk 3]
```diff
@@ -10,6 +10,66 @@
 #define PARSING_ALLOW_WITHOUT_ENFORCING_POLICY 2
 #define PARSING_ERROR -1
 
+#define MAX_EXTENSIONS 2
+
+static CALI_BPF_INLINE int parse_packet_ipv6(struct cali_tc_ctx *ctx, int iterations) {
+	__u8 next_header = ipv6_hdr(ctx)->nexthdr;
+
+	for (int i = 0; i < iterations; i++) {
+		if (skb_refresh_validate_ptrs(ctx, UDP_SIZE)) {
```
Comment (tomastigera) [inline felix/bpf-gpl/parsing.h:19]: we should know the value for each iteration, right? I can see where the verifier may have troubles. We should imo really se the get_data to get next_header

**Review by tomastigera — COMMENTED** [review]
i think we should merge this once we can get to the L4 header as the ->nh WDYT?

File: felix/bpf-gpl/parsing.h (hunk 1 above)
Comment (mazdakn) [reply felix/bpf-gpl/parsing.h]: Move this part to another PR.

File: felix/bpf-gpl/parsing.h (hunk 3 above)
Comment (mazdakn) [reply felix/bpf-gpl/parsing.h]: Address this in another PR.

File: felix/bpf-gpl/parsing.h (hunk 2 above)
Comment (mazdakn) [reply felix/bpf-gpl/parsing.h]: Fixed.

**Review by tomastigera — APPROVED** [review]

**Pull request merged**